firefox-ios/Client/Frontend/Browser/MainMenu/Views/MainMenuViewController.swift

```diff
@@ -128,16 +129,19 @@ class MainMenuViewController: UIViewController,
             }
 
             menuRedesignContent.onCalculatedHeight = { [weak self] height, isExpanded in
-                if #available(iOS 16.0, *), UIDevice.current.userInterfaceIdiom == .phone {
-                    let customDetent = UISheetPresentationController.Detent.custom { context in
-                        return height
-                    }
-                    if isExpanded {
-                        self?.sheetPresentationController?.animateChanges({
+                if let customHeight = self?.currentCustomMenuHeight, height > customHeight {
+                    self?.currentCustomMenuHeight = height
+                    if #available(iOS 16.0, *), UIDevice.current.userInterfaceIdiom == .phone {
```

Comment:
Just a note: Here it's better to use size classes for the case when we are in split screen on iPad and have the iPhone layout. Could be done in a follow up ticket too.